### ISSUE TITLE
Provider Social Connect

### ISSUE THREAD

**Issue by ClementLoridan** [strapi/strapi#1358]
**Node.js version**: v8.9.4

**npm version**: 5.6.0

**Strapi version**: 3.0.0-alpha.12.2

**Operating system**: OSX 10.12.6

I have a problem with Facebook Provider:

I use login-react example (localhost:3000) and my API Strapi (https://api.myexample.com in local).
I use https in local because Facebook developer doesn't accept URI url is not https

But the redirect is incorrect : http://localhost:3000/auth/facebook/callback?access_token=...

I think that redirect would be : /auth/facebook/callback?code=...

I change this in Auth.js (users-permissions)

```
server: {
        protocol: 'https',
        host: 'myexample.com'
      }
```

I don't know why ... Help please 

Thk

PS: Not working with other Provider.

**Comment by soupette** [comment]
Hi @ClementLoridan, I'm working on a new [boilerplate](https://github.com/strapi/strapi-examples/tree/update-boilerplate/create-react-app-sdk) (it's not on master yet) the uses the [`strapi-javascript-sdk`](https://github.com/strapi/strapi-sdk-javascript). I have recoded the basic authentication and I have just tested with Facebook which is working on my side.

Please make sure you followed these [steps](https://github.com/strapi/strapi-examples/blob/update-boilerplate/login-react/doc/fb_setup.md).

**Comment by ClementLoridan** [comment]
Merci de votre réponse !

J'ai identifié le problème :

Dans le services Providers.js du plugin "users-permissions", il y a 

```
if (!_.isEmpty(_.find(users, user => user.provider !== provider)) && advanced.unique_email) {
          return resolve([null, [{ messages: [{ id: 'Auth.form.error.email.taken' }] }], 'Email is already taken.']);
 } 
```

Je pense que cela devrait être ` user.provider === provider`.


Puis il y avait aussi une error ici :

```
if (!access_token) {
      return reject(null, {
        message: 'No access_token.'
      });
}
```

Qui je pense devrait être :

```
if (!access_token) {
      return reject([{null, {
        message: 'No access_token.'
      }]);
}
```

En tout les cas, super projet !! Je vais me mettre contributeur ce weekend ! 👍

**Comment by soupette** [comment]
Yes you'll need to disable the `One acccout per email address` in this [view](http://localhost:1337/admin/plugins/users-permissions/advanced)

**Comment by lauriejim** [comment]
Hello @ClementLoridan, can you please write all your messages in english please.
Will make this issue understandable by all the community.
Thank you :)

**Comment by lauriejim** [comment]
@ClementLoridan I don't understand the state of your issue. You look updating your comment instead of giving news. So the message/state history is hard to follow.

Can you give me the last news about it please.
